rust/src/write.rs

```diff
@@ -207,8 +224,22 @@ impl<'a, W: Write + Seek> Writer<'a, W> {
             }),
         )?;
 
+        // If both the `use_chunks` and `use_buffered_chunks` options are enabled set the chunk
+        // mode and pre-allocate the buffer. Checking both avoids allocating the temporary buffer
+        // if buffered chunks was enabled but chunks were not.
+        let chunk_mode = if opts.use_chunks && opts.use_buffered_chunks {
+            let buffer = opts
+                .chunk_size
+                .map(|size| Vec::with_capacity(size.try_into().unwrap_or(usize::MAX)))
```

Comment:
Why `usize::MAX`? Even with optimistic allocation, that's just going to fail due to insufficient virtual memory address space.

Reply:
Yeah wasn't super sure which way to go with this one. I figured semantically though if the consumer of the crate has chosen a chunk size larger than a `usize` then trying to allocate a buffer of that size would be somewhat expected behavior when buffering chunks. I'm not tied to this though so I'll happily return an error if you think that makes more sense.

Reply:
I think it makes sense to return a failed result here.